src/channel/ably/auth.ts

```diff
@@ -0,0 +1,139 @@
+import { beforeChannelAttach } from './attach';
+import { httpReqFunction, parseJwt, toTokenDetails } from './utils';
+import { SequentialAuthTokenRequestExecuter } from './token-request';
+import { AblyChannel } from '../ably-channel';
+import { AblyConnector } from '../../connector/ably-connector';
+import { AblyPresenceChannel } from '../ably-presence-channel';
+
+export class AblyAuth {
+
+    // TODO - Can be updated with request throttle, to send multiple request payload under single request
+    authRequestExecuter: SequentialAuthTokenRequestExecuter;
+    authEndpoint = '/broadcasting/auth';
+    authHost = typeof window != 'undefined' && window?.location?.hostname;
+    authPort = typeof window != 'undefined' && window?.location?.port;
+    authProtocol = typeof window != 'undefined' && window?.location?.protocol.replace(':', '');
+
+    httpReq = httpReqFunction();
+
+    authOptions = {
+        queryTime: true,
+        useTokenAuth: true,
+        authCallback: async (_, callback) => { // get token from tokenParams
+            try {
+                const { token } = await this.authRequestExecuter.request(null); // Replace this by network request to PHP server
+                const tokenDetails = toTokenDetails(token);
+                callback(null, tokenDetails);
+            } catch (error) {
+                callback(error, null);
+            }
+        }
+    }
+
+    requestToken = async (channelName: string, existingToken: string) => {
```

Comment:
Here and in a lot of other classes in this PR you're using function properties rather than actual class methods, I think we should stick to methods just to be more idiomatic

Reply:
Updated with actual class methods.

Reply:
Hey, as I thought and you might know. Class functions need to be binded externally through the constructor. We don't need to do it for arrow functions.

Also, somehow when I am calling a class method from another class and if I try to access this, it's giving me instances of other the calling class, rather than the one that's getting called. I think keeping it the old-fashioned way, should work fine ... Atleast for auth.ts. Class functions mess up this context ( does happen frequently in react and yes in js too )

Currently, I tested with new functions, and the change is breaking due to non-auto binding. So, reverting the internal class method changes.